code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/NotificationBuilder.kt

```diff
@@ -58,10 +60,10 @@ object NotificationBuilder {
             ?: throw NotificationConstructionFailedException("Application context is null, cannot build a notification.")
         val pushTemplateType =
             PushTemplateType.fromString(messageData[PushTemplateConstants.PushPayloadKeys.TEMPLATE_TYPE])
-
+        val notificationData = MapData(messageData)
```

Comment:
Now that you have consolidated the branching on `NotificationData`, how about we move the `when` branch to be common?

```
 fun constructNotificationBuilder(
        messageData: Map<String, String>,
        trackerActivityClass: Class<out Activity>?,
        broadcastReceiverClass: Class<out BroadcastReceiver>?
    ): NotificationCompat.Builder { 
    
   val context = ServiceProvider.getInstance().appContextService.applicationContext
            ?: throw NotificationConstructionFailedException("Application context is null, cannot build a notification.")
        val pushTemplateType =
            PushTemplateType.fromString(messageData[PushTemplateConstants.PushPayloadKeys.TEMPLATE_TYPE])
        val notificationData = MapData(messageData)
       return getNotificationBuilder(pushTemplateType, notificationData, trackerActivity, broadcastReceiver)
   }


    fun constructNotificationBuilder(
        intent: Intent,
        trackerActivityClass: Class<out Activity>?,
        broadcastReceiverClass: Class<out BroadcastReceiver>?
    ): NotificationCompat.Builder {
        val context = ServiceProvider.getInstance().appContextService.applicationContext
            ?: throw NotificationConstructionFailedException("Application context is null, cannot build a notification.")
        val extras = intent.extras ?: throw NotificationConstructionFailedException("Intent extras are null, cannot re-build the notification.")
        val pushTemplateType =
            PushTemplateType.fromString(intent.getStringExtra(PushTemplateConstants.PushPayloadKeys.TEMPLATE_TYPE))
        val intentData = IntentData(extras)
       
       return getNotificationBuilder(pushTemplateType, notificationData, trackerActivity, broadcastReceiver)
} 
```

```
fun getNotificationBuildeR(pushTemplateType: String, notificationData: NotificationData.....) {

  when(pushTemplateType) {
    PushTemplateType.BASIC ->. ....
    PushTemplateType.CAROUSEL -> ....
  }

}

  
```

Reply:
Another good call to simplify. I will take this in my next PR (trying to not overload this PR)